internal/apiresource/apiresource.go

```diff
@@ -38,17 +39,16 @@ const (
 
 // IAPIResource defines the interface to be defined for a new api resource
 type IAPIResource interface {
-	GetSupportedKinds() []string
-	CreateNewResources(ir irtypes.EnhancedIR, supportedKinds []string) []runtime.Object
+	getSupportedKinds() []string
```

Comment:
Why make the interface methods private?
We can make the interface private instead.

Reply:
The interface is still used by transformer which is outside this package.